homeassistant/components/script/__init__.py

```diff
@@ -401,7 +401,11 @@ async def _async_run(self, variables, context):
             # Prepare tracing the execution of the script's sequence
             script_trace.set_trace(trace_get())
             with trace_path("sequence"):
-                return await self.script.async_run(variables, context)
+                self.async_write_ha_state()
```

Comment:
Is this really needed here, it should only be needed in `_async_attach_triggers`?

Reply:
Yep, used the same initialization template in all 3 cases ...  I only needed to resolve a failure when incoming variables was None. I will remove and retest.

Reply:
Unit test passed ... code pushed.